client.go

```diff
@@ -879,7 +879,10 @@ func (c *Client) Invoke(ctx context.Context, root string, target string, m Invok
 	if err != nil {
 		return
 	}
-
+	// if invoke is empty, use default 'http'
+	if f.Invoke == "" {
+		f.Invoke = DefaultInvocationFormat
+	}
```

Comment:
I would go a step further and remove this condition from here because the `invoke` method already takes care of this logic.  It might just need to have the defaulting moved up to the very top of the function (before the verbose print statement).  The `invoke` method can expect to default to invoking via HTTP if not given an invocation method explicitly on the function or the message.